### PR TITLE
Run autopep8 on repo in aggressive mode

### PR DESCRIPTION
**Pull request by themalkolm**
Lets run autopep8 on repo in aggressive mode. It finds various minor
style issues and fixes type detection.

### CODE REVIEW

File: luigi/date_interval.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def __hash__(self):
         return hash(repr(self))
 
     def __cmp__(self, other):
-        if type(self) != type(other):
+        if not isinstance(self, type(other)):
```
Comment (Tarrasch) [inline luigi/date_interval.py]: `autopep8` converted this for you?

File: luigi/hdfs.py [hunk 2]
```diff
@@ -308,7 +308,7 @@ def get_bite(self):
         If Luigi has forked, we have a different PID, and need to reconnect.
         """
         if self.pid != os.getpid() or not self._bite:
-            client_kwargs = dict(filter(lambda (k, v): v is not None and v != '', {
+            client_kwargs = dict(filter(lambda k_v: k_v[1] is not None and k_v[1] != '', {
```
Comment (Tarrasch) [inline luigi/hdfs.py]: Is this really prettier?

**Comment by Tarrasch** [comment]
Nice!

File: luigi/hdfs.py (hunk 2 above)
Comment (themalkolm) [inline luigi/hdfs.py]: Nah, personally I think this lambda should be replaced with a simple operator.itemgetter(1) or lambda x: x[1] instead

But I left whatever autopep8 suggested

File: luigi/date_interval.py (hunk 1 above)
Comment (themalkolm) [inline luigi/date_interval.py]: Yep, how cool is that?

File: luigi/hdfs.py (hunk 2 above)
Comment (erikbern) [inline luigi/hdfs.py]: I really don't like treating tuples like arrays and using indices. Much nicer to unpack into variables

Comment (themalkolm) [inline luigi/hdfs.py]: Well, at least we all agree that autopep8 went for the weirdest solution.

**Pull request merged**